### PR TITLE
support nbd-log

### PR DESCRIPTION
**Pull request by Pating**
Signed-off-by: Xie Changlong <xiechanglong@cmss.chinamobile.com>

### CODE REVIEW

File: cli/nbd-cli.c [hunk 1]
```diff
@@ -118,11 +118,11 @@ static int load_our_module(void)
         state = kmod_module_get_initstate(mod);
         switch (state) {
         case KMOD_MODULE_BUILTIN:
-            nbd_out("Module '%s' is builtin\n", kmod_module_get_name(mod));
+            nbd_info("Module '%s' is builtin\n", kmod_module_get_name(mod));
```
Comment (xiubli) [inline cli/nbd-cli.c]: Let's make this as DEBUG info, and the same as the followings.

File: daemon/nbd-runner.c [hunk 2]
```diff
@@ -130,8 +130,8 @@ static void *nbd_map_svc_thread_start(void *arg)
     sin.sin_port = htons(iport);
 
     if (bind(listenfd, (struct sockaddr*)&sin, sizeof(struct sockaddr)) < 0) {
-        nbd_out("bind on port %d failed, %s\n", iport, strerror(errno));
-        nbd_out("will try to use port %d!\n", ++iport);
+        nbd_info("bind on port %d failed, %s\n", iport, strerror(errno));
+        nbd_info("will try to use port %d!\n", ++iport);
```
Comment (xiubli) [inline daemon/nbd-runner.c]: It will be better as WARN messages here.

File: daemon/nbd-svc-routines.c [hunk 3]
```diff
@@ -289,7 +289,7 @@ static int nbd_parse_from_json_config_file(void)
 
             strcpy(dev->bstore, key);
 
-            nbd_out("key: %s, type: %d, nbd: %s, maptime: %s, size: %ld, blksize: %ld, prealloc: %d, readonly: %d\n",
+            nbd_info("key: %s, type: %d, nbd: %s, maptime: %s, size: %ld, blksize: %ld, prealloc: %d, readonly: %d\n",
```
Comment (xiubli) [inline daemon/nbd-svc-routines.c]: Please make this as DEBUG messages.

File: cli/nbd-cli.c (hunk 1 above)
Comment (Pating) [reply cli/nbd-cli.c]: OK

File: daemon/nbd-runner.c (hunk 2 above)
Comment (Pating) [reply daemon/nbd-runner.c]: OK

File: daemon/nbd-svc-routines.c (hunk 3 above)
Comment (Pating) [reply daemon/nbd-svc-routines.c]: OK

File: utils/nbd-log.h [hunk 4]
```diff
@@ -28,59 +28,48 @@
...
+#define NBD_LOG_DEBUG_IO   (LOG_DEBUG + 1)	/* nbd io messages */
+
+struct nbd_device;
+
+void nbd_set_log_level(int level);
+unsigned int nbd_get_log_level(void);
+int nbd_setup_log(char *log_dir);
+void nbd_destroy_log(void);
+int nbd_make_absolute_logfile(char *path, const char *filename);
+int nbd_resetup_log_file(char *log_dir);
```
Comment (xiubli) [inline utils/nbd-log.h]: Please remove this useless API.

File: utils/nbd-log.h [hunk 5]
```diff
@@ -28,59 +28,48 @@
...
+#define NBD_LOG_DEBUG	LOG_DEBUG	/* debug-level messages */
+#define NBD_LOG_DEBUG_IO   (LOG_DEBUG + 1)	/* nbd io messages */
+
+struct nbd_device;
+
+void nbd_set_log_level(int level);
+unsigned int nbd_get_log_level(void);
+int nbd_setup_log(char *log_dir);
+void nbd_destroy_log(void);
+int nbd_make_absolute_logfile(char *path, const char *filename);
```
Comment (xiubli) [inline utils/nbd-log.h]: Please remove this and make it as static in the nbd-log.c file.

File: daemon/gluster.c [hunk 6]
```diff
@@ -35,7 +35,7 @@
 #include "utils.h"
 #include "nbd-common.h"
 
-#define NBD_GFAPI_LOG_FILE "/var/log/nbd-runner.log"
+#define NBD_GFAPI_LOG_FILE "/var/log/nbd-runner-glfs.log"
```
Comment (xiubli) [inline daemon/gluster.c]: Please include the "nbd-sysconfig.h" header file and then change this line as:
```
#define NBD_GFAPI_LOG_FILE NBD_LOG_DIR_DEFAULT"/nbd-runner-glfs.log"
```

File: utils/strlcpy.h [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#ifndef __STRLCPY_H
+#define __STRLCPY_H
+
+#include <sys/types.h>
+
+size_t strlcpy(char *dst, const char *src, size_t size);
```
Comment (xiubli) [inline utils/strlcpy.h:6]: And also let switch all the strncpy to strlcpy at the same time in the following files:
cli/nbd-cli-cmd.c
daemon/gluster.c
daemon/nbd-svc-routines.c

And comment it here for the reason:

For strncpy there is one warning:
Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated. The strlcpy code comes from the BSD and tcmu-runner project.

File: utils/nbd-log.h (hunk 4 above)
Comment (Pating) [reply utils/nbd-log.h]: OK

File: utils/nbd-log.h (hunk 5 above)
Comment (Pating) [reply utils/nbd-log.h]: OK

File: utils/strlcpy.h (hunk 7 above)
Comment (Pating) [reply utils/strlcpy.h:6]: OK

**Pull request merged**